pkg/kubelet/status/state/state_checkpoint_test.go

```diff
@@ -0,0 +1,105 @@
+package state
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
+)
+
+func Test_stateCheckpoint_storeState(t *testing.T) {
+	cache := NewStateMemory()
+	checkpointManager, _ := checkpointmanager.NewCheckpointManager(".")
+	checkpointName := "pod_state_checkpoint"
+	sc := &stateCheckpoint{
+		cache:             cache,
+		checkpointManager: checkpointManager,
+		checkpointName:    checkpointName,
+	}
+	var err error
+	type args struct {
+		podResourceAllocation PodResourceAllocation
+	}
+
+	tests := []struct {
+		name string
+		args args
+	}{}
+	suffix := []string{"Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "n", "u", "m", "k", "M", "G", "T", "P", "E", ""}
+	factor := []string{"1", "0.1", "0.03", "10", "100", "512", "1000", "1024", "700", "10000"}
+	for j, fact := range factor {
+		for i, suf := range suffix {
+			if (suf == "E" || suf == "Ei") && (fact == "1000" || fact == "10000") {
+				// when fact is 1000 or 10000, suffix "E" or "Ei", the quantity value is overflow
+				// see detail https://github.com/kubernetes/apimachinery/blob/95b78024e3feada7739b40426690b4f287933fd8/pkg/api/resource/quantity.go#L301
+				continue
+			}
+			tests = append(tests, struct {
+				name string
+				args args
+			}{
+				name: fmt.Sprintf("format %d - cpu %s%s", j*len(suffix)+i, fact, suf),
```

Comment:
"cpu" is misleading here as the same resource is also used for memory. How about `fmt.Sprintf("resource: %s%s", fact, suf)` or something similar?

Reply:
updated

Reply:
How format `j*len(suffix)+i` is going to be used?

Reply:
Sorry, I don't get your point. Could you please elaborate it?

Reply:
I think the number in the output doesn't bring a lot of value even not speaking about it correctness:
```
$ go test -v -run ^Test_stateCheckpoint_storeState$ k8s.io/kubernetes/pkg/kubelet/status/state | grep Test_ | tail -3
    --- PASS: Test_stateCheckpoint_storeState/#156_resource_-_10000T (0.00s)
    --- PASS: Test_stateCheckpoint_storeState/#157_resource_-_10000P (0.00s)
    --- PASS: Test_stateCheckpoint_storeState/#159_resource_-_10000 (0.00s)
```
And I'd propose to drop it, so the above output would look like this:
```
    --- PASS: Test_stateCheckpoint_storeState/resource_10000T (0.00s)
    --- PASS: Test_stateCheckpoint_storeState/resource_10000P (0.00s)
    --- PASS: Test_stateCheckpoint_storeState/resource_10000 (0.00s)
```
I'd even drop `resource`, but that's up to you.